packages/lesswrong/lib/editor/mentionsConfig.tsx

```diff
@@ -1,103 +1,126 @@
-    return response?.results?.[0];
-  };
+    }])
+    return response?.results?.[0]
+  }
 }
 
 async function fetchPostSuggestions(searchString: string) {
-  const search = initSearchForIndex<PostSearchHit>('Posts')
-  const searchResults = await search(searchString, ['title', 'slug', '_id']);
+  const search = initSearchForIndex<SearchPost>('Posts')
+  const searchResults = await search(searchString)
 
   return searchResults?.hits.map(hit => ({
+    type: 'Posts',
     id: markers.post + hit.title, //what gets displayed in the dropdown results, must have postMarker 
     link: `${linkPrefix}posts/${hit._id}/${hit.slug}`,
     text: hit.title,
-  }))
+    hit,
+  })) || []
 }
 
 async function fetchUserSuggestions(searchString: string) {
-  const search = initSearchForIndex<UserSearchHit>('Users')
-  const searchResults = await search(searchString, ['displayName', 'slug', '_id', 'username', 'groups', 'karma', 'createdAt', 'fullName']);
+  const search = initSearchForIndex<SearchUser>('Users')
+  const searchResults = await search(searchString)
 
   return searchResults?.hits.map(hit => {
     const displayName = markers.user + userGetDisplayName(hit)
     return ({
+      type: 'Users',
       id: displayName,
       // Query string is intended for later use in detecting the ping
       link: `${linkPrefix}users/${hit.slug}?${userMentionQueryString}`,
       text: displayName,
-      karma: hit.karma,
-      createdAt: new Date(hit.createdAt),
+      hit: {
+        ...hit,
+        displayName: '👤 ' + userGetDisplayName(hit),
+      },
     })
-  })
+  }) || []
+}
+
+async function fetchTagSuggestions(searchString: string) {
+  const search = initSearchForIndex<SearchTag>('Tags')
+  const searchResults = await search(searchString)
+
+  return searchResults?.hits.map(hit => ({
+    type: 'Tags',
+    id: markers.tag + hit.name,
+    link: `${linkPrefix}${taggingNamePluralSetting.get()}/${hit.slug}`,
+    text: hit.name,
+    hit,
+  })) || []
 }
 
-const renderUserItem = (item: { text: string, karma?: number, createdAt: Date }) => {
+const fetchMentionableSuggestions = async (searchString: string) => [
+  ...(await fetchUserSuggestions(searchString)),
+  ...(await fetchTagSuggestions(searchString)),
+  ...(await fetchPostSuggestions(searchString)),
+]
+
+interface MentionItem {
+  type: 'Users' | 'Posts' | 'Tags'
+  id: string
+  text: string
+  link: string
+  hit: SearchUser | SearchPost | SearchTag
+}
+
+const renderUserItem = (item: MentionItem) => {
   const itemElement = document.createElement('button')
+  itemElement.classList.add('ck-mention-item', 'ck-reset_all-excluded')
+  const root = createRoot(itemElement)
 
-  ReactDOM.render(<Components.UsersSearchAutocompleteHit {...item} name={item.text}/>, itemElement)
+  root.render(<Components.UsersSearchAutocompleteHit hit={item.hit as SearchUser}/>)
 
   return itemElement
 }
 
+const renderMentionItem = (
+  item: MentionItem,
+) => {
+  const renderer = {
+    Users: renderUserItem,
```

Comment:
Probably we should stick the user icon prefixing here too, for consistency?

Reply:
Here specifically is an awkward place bc this is a different type of thing then other renderers, I've played with having it in `renderUserItem` function bc I had a similar intuition, but it was a bit more awkward and I figured I'm overthinking things, so reverted that, but happy to bring that back

Reply:
Oh, nah, I didn't realize the user items were getting rendered in a substantially different way from the other two on a quick skim.  Maybe just a small comment to that effect